subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/worker/TestEventSerializer.java

```diff
@@ -171,7 +171,6 @@ public DefaultTestFailureSerializer(Serializer<Throwable> throwableSerializer) {
 
         @Override
         public DefaultTestFailure read(Decoder decoder) throws Exception {
-            Throwable rawFailure = throwableSerializer.read(decoder);
```

Comment:
What it handles cleanly is a failure to rehydrate a valid object that is properly transmitted (i.e. we get a ClassNotFound or something similar).  My understanding is that it is still potentially possible to have some sort of weird error where the object isn't valid or the stream is corrupt and this could blow up the stream before it finished fully reading the object.

Those are not things that should happen normally, but my thinking here was that if that _did_ happened, by moving this to the end of the stream, you would still at least get a test failure for _this_ test and the captured failure would be a clue as to what was happening.  It would be nice to have a test for that case, too, but that would be such a low level failure (i.e. creating/serializing invalid objects or corrupting the byte stream) it seemed like overkill.

I don't _think_ reordering the data in the test event protocol would affect anything.